### PR TITLE
[runtime] fix SIGSEGV inside gmmktime

### PR DESCRIPTION
**Pull request by Ramzeeset**

### CODE REVIEW

File: runtime/datetime/datetime_functions.cpp [hunk 1]
```diff
@@ -50,6 +51,24 @@ static time_t gmmktime(struct tm *tm) {
...
+static time_t gmmktime(struct tm *tm) {
+  dl::CriticalSectionGuard critical_section;
+  time_t deprecated_time = deprecated_gmmktime(tm);
+  time_t updated_time = updated_gmmktime(tm);
+  if (deprecated_time != updated_time) {
+    char * tz = getenv("TZ");
+    write_log_server_impl(ServerLog::Warning,
+                          "Updated gmmktime return different result. Updated value %ld. Deprecated value %ld. ENV TZ = \"%s\". "
+                          "Local tz name[0] = \"%s\", tz name[1] = \"%s\", timezone %ld, daylight %d",
+                          updated_time, deprecated_time, tz != nullptr ? tz : "null", tzname[0], tzname[1], timezone, daylight);
```
Comment (DrDet) [inline runtime/datetime/datetime_functions.cpp:67]: why don't use `log_server_warning`?

Comment (DrDet) [inline runtime/datetime/datetime_functions.cpp:67]: Can `tzname[0]` or `tzname[1]` be `nullptr`?

Comment (Ramzeeset) [reply runtime/datetime/datetime_functions.cpp]: fix

Comment (Ramzeeset) [reply runtime/datetime/datetime_functions.cpp]: I think no, but added checking to be sure

**Review by DrDet — APPROVED** [review]

**Pull request merged**